src/scripts/main.js

```diff
@@ -1,3 +1,352 @@
 'use strict';
 
-// write your code here
+const start = document.querySelector('.start');
+const scoreDisplay = document.querySelector('.game-score');
+const fieldRowAll = document.querySelectorAll('.field-row');
+const messageLose = document.querySelector('.message-lose');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+let probabilityCount = 0;
+let score = 0;
+let started = false;
+
+function getRandomInt(max) {
+  return Math.floor(Math.random() * max);
+};
+
+function createNewCell() {
+  if (!freeCell()) {
+    if (checkIfPlayerLose()) {
+      messageLose.classList.remove('hidden');
+    }
+
+    return;
+  }
+
+  let row = 0;
+  let cell = 0;
+
+  while (true) {
+    row = getRandomInt(4);
+    cell = getRandomInt(4);
+
+    if (board[row][cell] === 0) {
+      if (probabilityCount < 10) {
+        board[row][cell] = 2;
+        probabilityCount++;
+      } else {
+        board[row][cell] = 4;
+        probabilityCount = 0;
+      }
+
+      return;
+    }
+  }
+}
+
+function shiftRowLeft(row) {
+  const newRow = [];
+
+  for (let i = 0; i < row.length; i++) {
+    const element = row[i];
+
+    if (element !== 0) {
+      newRow.push(element);
+    }
+  }
+
+  for (let i = newRow.length; i < row.length; i++) {
+    newRow.push(0);
+  }
+
+  return newRow;
+}
+
+function checkAndAction(check) {
+  if (!check) {
+    createNewCell();
+    showDisplay();
+    checkIfPlayerWon();
+  }
+}
+
+function moveRight(check) {
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]).reverse();
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    for (let j = 1; j < board[i].length; j++) {
+      if (board[i][j - 1] === board[i][j]) {
+        board[i][j - 1] = 0;
+        board[i][j] = board[i][j] * 2;
+        score += board[i][j];
+      }
+    }
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]).reverse();
+  }
+
+  checkAndAction(check);
+}
+
+function moveLeft(check) {
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]);
+  }
+
+  for (let i = 0; i < board.length; i++) {
```

Comment:
Please replace the traditional loop with a functional programming method like 'forEach' for better readability.